direct/nn/conjgradnet/config.py

```diff
@@ -0,0 +1,32 @@
+# coding=utf-8
+# Copyright (c) DIRECT Contributors
+
+from dataclasses import dataclass
+from typing import Optional
+
+from direct.config.defaults import ModelConfig
+
+
+@dataclass
+class ConjGradNetConfig(ModelConfig):
+    num_steps: int = 8
+    image_init: str = "zeros"
+    no_parameter_sharing: bool = True
+    cg_tol: float = 1e-7
+    cg_iters: int = 10
+    cg_param_update_type: str = "FR"
+    denoiser_architecture: str = "resnet"
+    resnet_hidden_channels: int = 128
+    resnet_num_blocks: int = 15
+    resenet_batchnorm: bool = True
+    resenet_scale: Optional[float] = 0.1
+    unet_num_filters: Optional[int] = 32
+    unet_num_pool_layers: Optional[int] = 4
+    unet_dropout: Optional[float] = 0.0
+    didn_hidden_channels: Optional[int] = 16
+    didn_num_dubs: Optional[int] = 6
+    didn_num_convs_recon: Optional[int] = 9
+    conv_hidden_channels: Optional[int] = 64
+    conv_n_convs: Optional[int] = 15
+    conv_activation: Optional[str] = "relu"
```

Comment:
Not sure you want this to be a string. Enum's are good ideas at the time you are using this often. Same for your architecture. 
This I think I got from pytorch lightning:


`
class DirectEnum(str, Enum):
    """Type of any enumerator with allowed comparison to string invariant to cases."""

    @classmethod
    def from_str(cls, value: str) -> AhCoreEnum | None:
        statuses = cls.__members__.keys()
        for st in statuses:
            if st.lower() == value.lower():
                return cls[st]
        return None

    def __eq__(self, other: object) -> bool:
        if isinstance(other, Enum):
            _other = str(other.value)
        else:
            _other = str(other)
        return bool(self.value.lower() == _other.lower())

    def __hash__(self) -> int:
        # re-enable hashtable so it can be used as a dict key or in a set
        # example: set(LightningEnum)
        return hash(self.value.lower())
`

Reply:
I don't think you want these kind of strings :)
You're basically circumventing the typing.

I think the enum solution is good.